src/widgets/medications/medications.resource.tsx

```diff
@@ -7,14 +7,61 @@ import { Observable } from "rxjs";
 import { map, take, filter } from "rxjs/operators";
 import { OrderMedication } from "./medication-orders-utils";
 import dayjs from "dayjs";
+import { toOmrsDateString } from "../../utils/omrs-dates";
+import { FetchResponse } from "@openmrs/esm-api/dist/openmrs-fetch";
 
 const CARE_SETTING: string = "6f0c9a92-6f24-11e3-af88-005056821db0";
 const DURATION_UNITS_CONCEPT: string = "1732AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
 const NEW_MEDICATION_ACTION: string = "NEW";
 const REVISE_MEDICATION_ACTION: string = "REVISE";
 const DISCONTINUE_MEDICATION_ACTION: string = "DISCONTINUE";
 type PatientMedications = {
-  uuid: Number;
+  results: {
+    uuid: number;
+    action: string;
+    asNeed: boolean;
+    asNeededCondition?: string;
+    autoExpireDate: Date;
+    brandName?: string;
+    careSetting: { uuid: string; display: string };
+    commentToFulfiller: string;
+    dateActivated: Date;
+    dateStopped?: Date | null;
+    dispenseAsWritten: boolean;
+    dose: number;
+    doseUnits: { uuid: string; display: string };
+    dosingInstructions: string | null;
+    drug: {
+      name: string;
+      strenght: string;
+      concept: { uuid: string; display: string };
+    };
+    duration: number;
+    durationUnits: { uuid: string; display: string };
```

Comment:
Some of these subtypes are quite repetitive.

Example:

```ts
interface OpenMrsReference {
  uuid: string;
  display: string;
}
```

could be used in at least 12 properties. Now if that changes then we only would need to change a single interface, instead of all occurrences in this type.